container.go

```diff
@@ -53,6 +67,7 @@ type ContainerRequest struct {
 	Privileged   bool   // for starting privileged container
 
 	SkipReaper bool // indicates whether we skip setting up a reaper for this
+	err        error
```

Comment:
Is there a case to store the error in the struct? I would see it more convenient to return the error from the Validate method.

Wdyt?

Reply:
hmm I guess I don't have a strong one...I could change the `Validate` method to be something like this (pseudo code):
```go
var err error
for _, validateMethod := range []func() error {c.validateContextAndImage} {
    err = validateMethod()
    if err != nil {
        return err
    }
} 
return nil
```

then you could make the `validateContextAndImage` return an error, rather than modify the one on the struct.

Ultimately I wanted to have a clean way to validate the config, while making it easier to add more validation methods

Reply:
I like this second approach, as we do not alter the structure with the validation status, but as you said, I do not have a strong opinion on that. And not being a maintainer 😄 !!

For me it would be a 👍

Reply:
okay thanks for the feedback , I like the second approach better too, glad you brought this up! I can update the pull request at some time today!

Reply:
okay, I have updated the pull request with what we talked about, thanks again!